sdk/modelsrepository/Azure.Iot.ModelsRepository/src/Fetchers/LocalModelFetcher.cs

```diff
@@ -18,30 +18,27 @@ namespace Azure.Iot.ModelsRepository.Fetchers
     /// </summary>
     internal class LocalModelFetcher : IModelFetcher
     {
-        private readonly bool _tryExpanded;
         private readonly ClientDiagnostics _clientDiagnostics;
 
-        public LocalModelFetcher(ClientDiagnostics clientDiagnostics, ModelsRepositoryClientOptions clientOptions)
+        public LocalModelFetcher(ClientDiagnostics clientDiagnostics)
         {
             _clientDiagnostics = clientDiagnostics;
-            _tryExpanded = clientOptions.DependencyResolution == DependencyResolutionOption.TryFromExpanded;
         }
 
-        public Task<FetchResult> FetchAsync(string dtmi, Uri repositoryUri, CancellationToken cancellationToken = default)
+        public Task<FetchResult> FetchAsync(string dtmi, Uri repositoryUri, DependencyResolutionOption resolutionOption, CancellationToken cancellationToken = default)
```

Comment:
Observation: Enums are not nullable by nature, so we don't need null check for them here.